opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/QueryStringQuery.java

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * Class for Lucene query that builds the query_string query.
+ */
+public class QueryStringQuery extends RelevanceQuery<QueryStringQueryBuilder> {
+  /**
+   *  Default constructor for QueryString configures how RelevanceQuery.build() handles
+   * named arguments.
+   */
+  public QueryStringQuery() {
+    super(ImmutableMap.<String, QueryBuilderStep<QueryStringQueryBuilder>>builder()
+        .put("analyzer", (b, v) -> b.analyzer(v.stringValue()))
+        .put("allow_leading_wildcard", (b, v) ->
+            b.allowLeadingWildcard(Boolean.parseBoolean(v.stringValue())))
+        .put("analyze_wildcard", (b, v) ->
+            b.analyzeWildcard(Boolean.parseBoolean(v.stringValue())))
+        .put("auto_generate_synonyms_phrase_query", (b, v) ->
+            b.autoGenerateSynonymsPhraseQuery(Boolean.parseBoolean(v.stringValue())))
+        .put("boost", (b, v) -> b.boost(Float.parseFloat(v.stringValue())))
+        .put("default_operator", (b, v) ->
+            b.defaultOperator(Operator.fromString(v.stringValue())))
+        .put("enable_position_increments", (b, v) ->
+            b.enablePositionIncrements(Boolean.parseBoolean(v.stringValue())))
+        .put("fuzziness", (b, v) -> b.fuzziness(Fuzziness.build(v.stringValue())))
+        .put("fuzzy_rewrite", (b, v) -> b.fuzzyRewrite(v.stringValue()))
+        .put("escape", (b, v) -> b.escape(Boolean.parseBoolean(v.stringValue())))
+        .put("fuzzy_max_expansions", (b, v) ->
+            b.fuzzyMaxExpansions(Integer.parseInt(v.stringValue())))
+        .put("fuzzy_prefix_length", (b, v) ->
+            b.fuzzyPrefixLength(Integer.parseInt(v.stringValue())))
+        .put("fuzzy_transpositions", (b, v) ->
+            b.fuzzyTranspositions(Boolean.parseBoolean(v.stringValue())))
+        .put("lenient", (b, v) -> b.lenient(Boolean.parseBoolean(v.stringValue())))
+        .put("max_determinized_states", (b, v) ->
+            b.maxDeterminizedStates(Integer.parseInt(v.stringValue())))
+        .put("minimum_should_match", (b, v) -> b.minimumShouldMatch(v.stringValue()))
+        .put("quote_analyzer", (b, v) -> b.quoteAnalyzer(v.stringValue()))
+        .put("phrase_slop", (b, v) -> b.phraseSlop(Integer.parseInt(v.stringValue())))
+        .put("quote_field_suffix", (b, v) -> b.quoteFieldSuffix(v.stringValue()))
+        .put("rewrite", (b, v) -> b.rewrite(v.stringValue()))
+        .put("type", (b, v) -> b.type(MultiMatchQueryBuilder.Type.parse(v.stringValue(),
+            LoggingDeprecationHandler.INSTANCE)))
+        .put("tie_breaker", (b, v) -> b.tieBreaker(Float.parseFloat(v.stringValue())))
+        .put("time_zone", (b, v) -> b.timeZone(v.stringValue()))
+        .build());
+  }
+
+  /**
+   * Override base build function for multi-field query support.
+   * @param func  function : 'query_string' function
+   * @return : QueryBuilder for query_string query
+   */
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
```

Comment:
We may consider pull up this method to base class or some utils. Because the logic seems duplicate in several class with only 1 line different that creates different query builder.

Reply:
Yes.  We have an action/PR coming to refactor the query builder.  It'll be coming but is currently stalled because of vacations.

Reply:
Sounds good. Thanks!